File: fetch_data.py
from flask import Flask, jsonify
import pymysql

app = Flask(__name__)

# Database connection details
servername = "localhost"
username = "your_username"
password = "your_password"
dbname = "your_database_name"

@app.route('/data')
def get_data():
    try:
        # Connect to the database
        conn = pymysql.connect(host=servername, user=username, password=password, database=dbname)
        cursor = conn.cursor()

        # SQL query
        sql = "SELECT id, waterlevel, temperature, humidity, timestamp FROM ESPDATA ORDER BY id DESC"

        # Execute the query
        cursor.execute(sql)

        # Fetch all rows
        rows = cursor.fetchall()

        # Close the cursor and connection
        cursor.close()
        conn.close()

        # Format data as a list of dictionaries
        data = []
        for row in rows:
            data.append({
                "id": row[0],
                "waterlevel": row[1],
                "temperature": row[2],
                "humidity": row[3],
                "timestamp": row[4]
            })

        return jsonify({"data": data})

    except Exception as e:
        return jsonify({"error": str(e)})

if __name__ == '__main__':
    app.run(debug=True)

File: fetch_data.php
<?php
$servername = "sql12.freesqldatabase.com";
$dbname = "sql12672580";
$username = "sql12672580";
$password = "KetWXbpE2z";

// Create connection
$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("Connection failed: " . $conn->connect_error);
}

$sql = "SELECT ID, RECORDED_DATA, WATERLEVEL, TEMPERATURE, HUMIDITY FROM ESPDATA";
$result = $conn->query($sql);

$data = array();

if ($result->num_rows > 0) {
    // Output data of each row
    while($row = $result->fetch_assoc()) {
        $data[] = $row;
    }
} else {
    $data[] = array("message" => "No results found");
}

$conn->close();

header('Content-Type: application/json');
echo json_encode($data);
?>
